### PR TITLE
754 sso saml support

### PR DESCRIPTION
**Pull request by brianbolt**

### CODE REVIEW

File: modules/ServerAPI/src/server/routes/AuthorRoutes.coffee [hunk 1]
```diff
@@ -805,32 +815,36 @@ checkEmailIsUnique = (author, callback) ->
...
 	callback null, author, systemRoles
 
-
-diffSystemRolesWithSaved = (userName, systemRoles, savedSystemRoles, callback) ->
-	rolesToAdd = _.filter systemRoles, (sysRole) ->
-		!(_.findWhere savedSystemRoles, (id: sysRole.id))?
-	rolesToDelete = _.filter savedSystemRoles, (savedSysRole) ->
-		!(_.findWhere systemRoles, (id: savedSysRole.id))?
-	fetchSystemRoles rolesToAdd, (err, rolesToAdd) ->
+diffSystemRolesWithSaved = (userName, systemRoles, savedSystemRoles, matchKeys, callback) ->
```
Comment (bffrost) [inline modules/ServerAPI/src/server/routes/AuthorRoutes.coffee:842]: I see, matchKeys makes this more flexible

**Review by bffrost — CHANGES_REQUESTED** [review]

File: modules/ServerAPI/src/server/CustomerSpecificServerFunctions.coffee [hunk 2]
```diff
@@ -148,7 +149,143 @@ exports.isUserAdmin = (user) ->
...
+	else
+		author = 
+			firstName: profile.firstName
+			lastName: profile.lastName
+			emailAddress: profile.email
+			userName: profile[userNameAttribute]
+			version: 0
+			enabled: true
+			locked: false
+			password: 'saml managed password'
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply modules/ServerAPI/src/server/CustomerSpecificServerFunctions.coffee:231]: Agreed, I will set it to null as it doesn't makes sense to fill it in.   In either case, it wouldn't allow login as the password needs to be hashed for it to work.

File: modules/ServerAPI/src/server/routes/AuthorRoutes.coffee [hunk 3]
```diff
@@ -101,7 +101,7 @@ exports.getAuthorByUsernameInternal = (username, callback) ->
 			json: true
 		, (error, response, json) =>
 			if !error && response.statusCode == 200
-				callback json, 200
+				callback json, 500
```
Comment (bffrost) [inline modules/ServerAPI/src/server/routes/AuthorRoutes.coffee:104]: Sorry why's this a 500 now? This looks like the successful response

**Review by bffrost — APPROVED** [review]
LGTM! Super excited to try this out.

**Pull request merged**